core/src/main/scala/org/apache/spark/deploy/master/ApplicationInfo.scala

```diff
@@ -38,11 +38,11 @@ private[spark] class ApplicationInfo(
   extends Serializable {
 
   @transient var state: ApplicationState.Value = _
-  @transient var executors: mutable.HashMap[Int, ExecutorDesc] = _
-  @transient var removedExecutors: ArrayBuffer[ExecutorDesc] = _
-  @transient var coresGranted: Int = _
+  @transient private[master] var executors: mutable.HashMap[Int, ExecutorDesc] = _
+  @transient private[master] var removedExecutors: ArrayBuffer[ExecutorDesc] = _
+  @transient private[master] var coresGranted: Int = _
   @transient var endTime: Long = _
-  @transient var appSource: ApplicationSource = _
+  @transient private[master] var appSource: ApplicationSource = _
```

Comment:
I think this might be going a little too far. It's probably OK for all of these to just be `private[deploy]`